app/src/main/java/gateway/controller/CtrlFileList.java

```diff
@@ -0,0 +1,100 @@
+package gateway.controller;
+
+import gateway.config.Config;
+import gateway.services.ServiceAuth;
+import gateway.soap.request.ReqFileList;
+import gateway.soap.response.File;
+import gateway.soap.response.ResFileList;
+import gateway.soap.response.ResStatus;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.UUID;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class CtrlFileList
+{
+	public static ResFileList file_list (ReqFileList args)
+	{
+		// Create a new instance of ResFileList to store the results
+		ResFileList resFileList = new ResFileList ();
+		UUID userUUID;
+		String url;
+
+		// User authentication
+		ResStatus resAuth = ServiceAuth.authenticate (args.token);
+		if (resAuth.error) {
+			return ResStatus.downCast (ResFileList.class, resAuth);
+		}
+
+		// Get the user's UUID from the authentication token
+		userUUID = UUID.fromString (ServiceAuth.tokenGetClaim (args.token, "uuid"));
+
+		// Build the URL for the request.
+		if (args.location == null || args.location.equals (new UUID (0L, 0L))) {
+			// If no location is specified, build the URL without a parentUUID parameter
+			url = Config.getMetadataBaseUrl () + "/files/list/" + userUUID;
+		} else {
+			// If a location is specified, include the parentUUID parameter
+			url = Config.getMetadataBaseUrl () + "/files/list/" + userUUID +
+				  "?parentUUID=" + args.location;
+		}
+
+		try {
+			// Make an HTTP GET request
+			HttpResponse<String> response = HttpClient.newHttpClient ().send (
+				HttpRequest.newBuilder ().uri (URI.create (url)).GET ().build (),
+				HttpResponse.BodyHandlers.ofString ());
+
+			JSONObject responseBody = new JSONObject (response.body ());
+			resFileList.code = response.statusCode ();
+
+			if (resFileList.code == 200) {
+				// If the response code is 200, process the received files
+				JSONArray filesArray = responseBody.getJSONArray ("files");
+				File[] files = new File[filesArray.length ()];
+
+				for (int i = 0; i < filesArray.length (); i++) {
+					// Process each file in the response JSON
+					JSONObject fileObject = filesArray.getJSONObject (i);
+
+					// Extract file information
+					String fileuuid = fileObject.getString ("uuid");
+					String fileType = fileObject.getString ("fileType");
+					String fileName = fileObject.getString ("name");
+					String fileExtension =
+						fileObject.isNull ("extension") ? null : fileObject.getString ("extension");
+
+					// Create a File object and assign it the file information
+					File file = new File ();
+
+					file.uuid = UUID.fromString (fileuuid);
+					file.name = (fileExtension == null) ? fileName : fileName + "." + fileExtension;
```

Comment:
Pass the name as is, do not append the extension at the end. For example, if a user uploads an image called `picture.png` the end result would be something like `picture.png.image/png`. The file extension could already be part of the file name and the extension we store is not in a format in which can be appended to the file name.

Reply:
I'm having the same problem. I thought we were saving the extension separately from the name.

We had discussed this previously. If the extension is not separated from the name, users can remove it when renaming the file.

Reply:
Resolved. Users are responsible for their files, and the `.ext` is considered part of the name; they can remove it if they want to.